ThemingCodelab/app/build.gradle

```diff
@@ -22,37 +23,126 @@ plugins {
 android {
     compileSdkVersion 33
     defaultConfig {
-        applicationId 'com.codelab.theming'
+        applicationId 'com.example.jetnews'
         minSdkVersion 21
         targetSdkVersion 33
         versionCode 1
         versionName '1.0'
+        vectorDrawables.useSupportLibrary = true
+        testInstrumentationRunner 'androidx.test.runner.AndroidJUnitRunner'
+    }
+
+    signingConfigs {
+        // We use a bundled debug keystore, to allow debug builds from CI to be upgradable
+        debug {
+            storeFile rootProject.file('debug.keystore')
+            storePassword 'android'
+            keyAlias 'androiddebugkey'
+            keyPassword 'android'
+        }
     }
+
+    buildTypes {
+        debug {
+            signingConfig signingConfigs.debug
+        }
+
+        release {
+            minifyEnabled true
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+
+    }
+    testOptions {
+        unitTests {
+            includeAndroidResources = true
+        }
+    }
+
+    // Tests can be Robolectric or instrumented tests
+    sourceSets {
+        String sharedTestDir = 'src/sharedTest/java'
+        test {
+            java.srcDir sharedTestDir
+        }
+        androidTest {
+            java.srcDir sharedTestDir
+        }
+    }
+
     compileOptions {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
-    kotlinOptions {
-        jvmTarget = '1.8'
-        allWarningsAsErrors = true
-    }
+
     buildFeatures {
-         compose true
+        compose true
     }
+
     composeOptions {
         kotlinCompilerExtensionVersion '1.3.2'
     }
+
+    packagingOptions {
+        excludes += "/META-INF/AL2.0"
+        excludes += "/META-INF/LGPL2.1"
+    }
 }
 
 dependencies {
     def composeBom = platform('androidx.compose:compose-bom:2023.01.00')
     implementation(composeBom)
+    testImplementation(composeBom)
+    androidTestImplementation(composeBom)
 
-    implementation 'androidx.core:core-ktx:1.9.0'
+    implementation "androidx.compose.runtime:runtime"
     implementation "androidx.compose.ui:ui"
-    implementation "androidx.compose.material:material"
```

Comment:
why remove them from here? material is in bom

Reply:
updated to bom